### PR TITLE
Update ForwardingRule and GlobalForwardingRule to add pscConnectionId…

### PR DESCRIPTION
**Pull request by muresan**
… and pscConnectionStatus attributes

Fixes https://github.com/hashicorp/terraform-provider-google/issues/10588

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [ ] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-the-terraform-providers), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [ ] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/third_party/terraform/tests) (for handwritten resources or update tests).
- [ ] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
compute: add psc_connection_id and psc_connection_status to google_compute_forwarding_rule and google_compute_global_forwarding_rule
```

### CODE REVIEW

**Comment by ScottSuarez** [comment]
could you follow the steps to sign the CLA?

**Review by ScottSuarez — CHANGES_REQUESTED** [review]

File: mmv1/products/compute/api.yaml [hunk 1]
```diff
@@ -4114,6 +4114,23 @@ objects:
...
+        description: |
+          The PSC connection status, can be one of the following: ACCEPTED, CLOSED, PENDING,
+          REJECTED, STATUS_UNSPECIFIED, UNDEFINED_PSC_CONNECTION_STATUS.
+        values:
+          - :ACCEPTED
+          - :CLOSED
+          - :PENDING
+          - :REJECTED
+          - :STATUS_UNSPECIFIED
+          - :UNDEFINED_PSC_CONNECTION_STATUS
```
[the remark this answers is not in the document]
Comment (muresan) [reply mmv1/products/compute/api.yaml]: Does not look like it is settable, documentation is incomplete, the console has no options to set this field. Changed to String and output: true.

**Comment by muresan** [comment]
> could you follow the steps to sign the CLA?

CLA shows signed on my end (and the bot also shows OK).

File: mmv1/products/compute/api.yaml [hunk 2]
```diff
@@ -4114,6 +4114,17 @@ objects:
           The internal fully qualified service name for this Forwarding Rule.
           This field is only used for INTERNAL load balancing.
         output: true
+      - !ruby/object:Api::Type::String
+        name: 'pscConnectionId'
+        description: |
+          [Output Only] The PSC connection id of the PSC Forwarding Rule in uint64 format.
```
Comment (ScottSuarez) [inline mmv1/products/compute/api.yaml:4120]:
```suggestion
          The PSC connection id of the PSC Forwarding Rule in uint64 format.
```

Remove [Output Only], documentation automation handles this

File: mmv1/products/compute/api.yaml [hunk 3]
```diff
@@ -4491,6 +4502,17 @@ objects:
           addressType of INTERNAL, only "all-apis" and "vpc-sc" are valid.
         update_verb: :POST
         update_url: 'projects/{{project}}/global/forwardingRules/{{name}}/setTarget'
+      - !ruby/object:Api::Type::String
+        name: 'pscConnectionId'
+        description: |
+          [Output Only] The PSC connection id of the PSC Forwarding Rule in uint64 format.
```
Comment (ScottSuarez) [inline mmv1/products/compute/api.yaml:4508]:
```suggestion
          The PSC connection id of the PSC Forwarding Rule in uint64 format.
```

**Review by ScottSuarez — CHANGES_REQUESTED** [review]

File: mmv1/products/compute/api.yaml (hunk 3 above)
Comment (muresan) [reply mmv1/products/compute/api.yaml]: Updated

**Comment by muresan** [comment]
> So unfortunately it looks like these resources are generated by the DCL which is a library that our team doesn't directly own. Thus we are unable to surface these fields :/
> 
> If you look at the generated providers it seems only the documentation is getting updated.

I got here from https://github.com/hashicorp/terraform-provider-google/pull/11767 which lead to https://github.com/GoogleCloudPlatform/declarative-resource-client-library/pull/11 which lead me to create this PR. As you can see from this comment @megan07  https://github.com/hashicorp/terraform-provider-google/pull/11767#issuecomment-1137459731 magic-modules was mentioned as the source for DCL (my assumption here is that by DCL you mean GoogleCloudPlatform/declarative-resource-client-library, if not please correct me)

What are the options ? since the https://github.com/GoogleCloudPlatform/declarative-resource-client-library/blob/main/docs/contributing.md for declarative-resource-client-library clearly offers to options.

Chicken and Egg problem ?

**Comment by muresan** [comment]
> If you look at the generated providers it seems only the documentation is getting updated.

I ran locally the command to generate the terraform code:
```
make terraform VERSION=ga OUTPUT_PATH="$GOPATH/src/github.com/hashicorp/terraform-provider-google"
``
and it _did_ generate the code but at the very end there was a warning and then it reverted back all the code. If you execute `git status` in the terraform provider source code before it completes but after it generates Compute you will see the code there. That's why I did not check above the Generate Terraform option, assumed my local ruby (or something) had issues.

```
12:32 $ git status|grep  compute_ | grep -v website
	modified:   google/resource_compute_forwarding_rule.go
	modified:   google/resource_compute_forwarding_rule_sweeper_test.go
	modified:   google/resource_compute_global_forwarding_rule.go
	modified:   google/resource_compute_global_forwarding_rule_sweeper_test.go
```
Here are the generated files: https://gist.github.com/muresan/7c1aca765407eb9fb6611e1c02c8aaab
They "disappear" at the end, the output form the run is also in the gist

Not sure how to debug this further

**Comment by jamesagarside** [comment]
Thank you @muresan for creating this pull request. I am very keen for this functionality as its currently a blocker on a project of mine. 
Is there anything I can help with?

**Comment by ScottSuarez** [comment]
We're working to add this to the DCL. It should make it's way downstream within the next few weeks.

**Comment by muresan** [comment]
> We're working to add this to the DCL. It should make it's way downstream within the next few weeks.

Looks like Private Service Connect structures were recently added to the DCL: https://github.com/GoogleCloudPlatform/declarative-resource-client-library/commit/785f52764dd60f0beb449f590acbb3a4c7c16639 , v1.11.0 has the required resources, what would be the next steps ?

Thanks

**Comment by ScottSuarez** [comment]
The next step would be updating the DCL package by running `make upgrade dcl`

Looks like the package has already been updated 
https://github.com/GoogleCloudPlatform/magic-modules/pull/6124

**Comment by muresan** [comment]
> The next step would be updating the DCL package by running `make upgrade dcl`
> 
> Looks like the package has already been updated #6124

Closing this PR as it's useless now. Curious why this change was not be merged since it's identical in functinality to what was merged.

**Comment by razine-bensari** [comment]
Is it possible to reopen this ?
I think this feature is quite useful.
Why was it closed ?

**Pull request closed without merging**